include/h2o/socket.h

```diff
@@ -394,6 +394,10 @@ int h2o_socket_ebpf_init_key(h2o_ebpf_map_key_t *key, void *sock);
 void h2o_socket__write_pending(h2o_socket_t *sock);
 void h2o_socket__write_on_complete(h2o_socket_t *sock, int status);
 
+SSL *h2o_ssl_get_ssl(h2o_socket_t *sock);
+SSL_CTX *h2o_ssl_get_ssl_ctx(h2o_socket_t *sock);
+struct st_ptls_t *h2o_ssl_get_socket_ptls(h2o_socket_t *sock);
```

Comment:
Thank you for the PR.

Unfortunately, I'm not sure if providing direct access to _underlying implementations_ of a socket is going to help us in the long run. For obtaining properties, we have functions like `h2o_socket_get_ssl_cipher_bits` that work regardless of what the stack in action is.

FWIW, the HTTP level, we also provide HTTP-version- / implementation-agnostic APIs through `h2o_conn_callbacks_t`.

I wonder if in-house branches can use these APIs whenever possible, or extend these version- / implementation-agnostic callbacks, rather than creating shortcuts. I know creating shortcuts is easy for the short-term, but...

Reply:
OK, thank you for this explanation.  Will close this PR then.